pkgs/development/lua-modules/overrides.nix

```diff
@@ -6,10 +6,12 @@ with super;
   #### manual fixes for generated packages
   ##########################################3
   bit32 = super.bit32.override({
-    disabled = !isLua51;
-    # Theoretically works with luajit, but it does redefine at least one Lua
-    # 5.2 function that Luajit 2.1 provides, see:
-    # https://github.com/LuaJIT/LuaJIT/issues/325
+    # Small patch in order to no longer redefine a Lua 5.2 function that Luajit
+    # 2.1 also provides, see https://github.com/LuaJIT/LuaJIT/issues/325 for
+    # more
+    patches = [
```

Comment:
can't we patch just for isLua51 then, like `patches = optionals isLua51 ./bit32.patch;`.
It would limit potential issues to lua51 (some damage control).

Reply:
(Re)defining these macros is quite safe, as their exact implementation is even [documented](https://www.lua.org/manual/5.3/manual.html#luaL_newlib) (and doesn't change AFAIK).